### DISCUSSION TITLE
Query grain state from storage

### DISCUSSION THREAD

**Discussion by gschuager** [dotnet/orleans#9219] [Q&A] [answered]
I have several grains with GUID ids that persist state to the OrleansStorage table using the ADO.NET provider.
I am wondering how can I locate the row corresponding to a given grain based on its type and id... I see the grainidhash, grainidn0, grainidn1 and grainidextension columns but I couldn't find any docs related to how to map a given GUID to this columns for querying.

**Comment by gfix** [comment, marked as answer]
You can see how the `Guid` is transformed into `GrainIdN0`, `GrainIdN1` and `GrainIdExtension` [here](https://github.com/dotnet/orleans/blob/main/src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/AdoGrainKey.cs#L37). The  `GrainIdHash` is created [here](https://github.com/dotnet/orleans/blob/main/src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/AdoNetGrainStorage.cs#L348).
